library/Faktory/Worker.hs

```diff
@@ -77,61 +76,90 @@ instance ToJSON FailPayload where
   toJSON = genericToJSON $ aesonPrefix snakeCase
   toEncoding = genericToEncoding $ aesonPrefix snakeCase
 
-forkWithThrowToParent :: Worker () -> Worker ThreadId
-forkWithThrowToParent action = do
-  parent <- liftIO myThreadId
-  workerConfig <- ask
-  liftIO $ forkIO $ (flip runReaderT workerConfig . runWorkerM $ action) `catchAny` \err -> throwTo parent err
-
-createWorker
-  :: HasCallStack
-  => Settings
+untilM_ :: Monad m => m Bool -> m a -> m ()
+untilM_ predicate action = do
+  result <- predicate
+  unless
+    result
+    ( do
+        void action
+        untilM_ predicate action
+    )
+
+-- | Creates a new faktory worker, @'action'@ is ran with @'WorkerConfig'@ before
+-- polling begins. Jobs received are passed to @'handler'. The worker's
```

Comment:
[Do] Did you miss the closing `@` on 'handler'?

Reply:
yup sounds like me, I actually have no clue what these do either so i need to do some more reading lol